### ISSUE TITLE
using lxc as opposed to docker for development

### ISSUE THREAD

**Issue by muralisrini** [openblockchain/obc-dev-env#38]
This issue is opened to discuss the work started by @ghaskins in https://github.com/ghaskins/obc-dev-env/tree/lxc-clones.

Currently development model uses docker image to launch obc-peer. A typical development workflow 
  1. vagrant ssh - the vagrant env points to obc-peer files on host
  2. build "openchain" obc-peer docker image
  3. docker run ... openchain -  effectively simulate a new node on the network with its own db and ip address

An alternative proposed in https://github.com/ghaskins/obc-dev-env/tree/lxc-clones has the following workflow
  1. vagrant ssh - the vagrant env points to obc-peer files on host
  2. lxc-execute to clone the current shell - share obc-peer code (so all the way from host) but create local copies for other env such as /var/openchain/db. However, just like the docker approach, effectively simulate a new node on the network with its own db and ip address

Basically, the lxc approach removes the need to create a docker image. This helps with rapid development turnaround for both chaincode developers and OBC developers
  *  chaincode developers don't have to build the peer image with their chaincode under development
  * OBC developers don't have to build peer image to pick up OBC code changes for dev/test activities. 

It may be possible to partially work around above issues by having vagrant and docker container share some files, but one cannot completely overcome need to build  the image.

On a related point, docker compose is the other  development environment we'd have to keep in mind as we discuss this. (@corecode  @mastersingh24)

**Comment by ghaskins** [comment]
@muralisrini with the heavy investment in docker-compose and the emerging support for a faster docker environment (via PR #49, etc), this experiment is probably moot and can be closed.  Would you agree?

**Comment by muralisrini** [comment]
@ghaskins : yes, that sounds about right... there were aspects that were cool (sigh). Will close.
